iroha/src/sumeragi.rs

```diff
@@ -1652,6 +1652,30 @@ pub mod config {
         }
     }
 
+    /// `SumeragiConfiguration` provides an ability to define parameters such as `BLOCK_TIME_MS`
+    /// and list of `TRUSTED_PEERS`.
+    #[derive(Clone, Debug, Deserialize)]
+    #[serde(rename_all = "UPPERCASE")]
+    pub struct TrustedPeers {
+        /// Optional list of predefined trusted peers.
+        #[serde(default)]
+        pub peers: BTreeSet<PeerId>,
+    }
+
+    impl TrustedPeers {
+        /// Load trusted peers variables from a json *pretty* formatted file.
+        pub fn from_path<P: AsRef<Path> + Debug>(path: P) -> Result<TrustedPeers, String> {
```

Comment:
It would be better not to use `String` as error at all now.

Reply:
Thanks! so what I should use? sorry

Reply:
use `iroha_error::Result`! It actually is an alias for `Result<T, iroha_error::Error>`. You can convert any error to it

Reply:
There is also an `error!` macros which takes a string and wraps it in error. You can use it or make an enum with variants of your error and `#[derive(Error)]` for them. Check `iroha_error` crate and its docs for more examples.